airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2076,12 +2081,11 @@ def run_query(
         labels = labels or self.labels
         schema_update_options = list(schema_update_options or [])
 
+        priority = priority or self.priority
+
         if time_partitioning is None:
             time_partitioning = {}
 
-        if location:
-            self.location = location
-
```

Comment:
I'm not sure if this is used in some other way to propagate location to the instance attribute. It isn't the right thing to do but at the same time, I'm worried that an operator was relying on this functionality - especially given the fact that there was a test around this. Can you check if callers of run_query were previously relying on it?

Reply:
> > I think the ideal fix is that run_query() is deprecated and cursor should instead call insert_job(), however, need the former to parse api_resource_configs from hook and build this into configuration for insert_job()
> 
> This makes sense to me. You are suggesting we replace [this call to run_query](https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/hooks/bigquery.py#L2717) to use [insert_job](https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/hooks/bigquery.py#L1507) instead right? 

[yup](https://github.com/apache/airflow/pull/30655/commits/5dabb5b8427927474bc43cd38da9d24d94412b71)

> Would it be possible to refactor the api_resource_configs parsing done in run_query and call that within [execute](https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/hooks/bigquery.py#L2708)?

yeah, just wasn't sure how much to put in one PR

btw, is there a way to test changes kinda like `pip install -e .`? I couldn't figure out how to 'install' the provider and instead modifying the code in python site-packages

Reply:
i was just checking what other calls go to `run_query()` and in hooks/bigquery.py theres only 1 other call from deprecated [`BigQueryCursor.run_query`](https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/hooks/bigquery.py#L2655)

in operators/bigquery.py are 2 calls from within the deprecated `BigQueryExecuteQueryOperator` class just passing through values for [`execute()`](https://github.com/apache/airflow/blob/29fb38cfe3b0f1e54b48e7cc5786ab1c15ab8425/airflow/providers/google/cloud/operators/bigquery.py#L1097)

So getting rid of the call from `cursor.execute()` should remove calls to run_query() from non-deprecated functions

(i checked all references of `run_query`  within airflow/providers/google/cloud/)